pkg/logs/input/file/tailer_windows.go

```diff
@@ -0,0 +1,96 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+// +build windows
+
+package file
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/DataDog/datadog-agent/pkg/logs/decoder"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// setup sets up the file tailer
+func (t *Tailer) setup(offset int64, whence int) error {
+	fullpath, err := filepath.Abs(t.path)
+	if err != nil {
+		return err
+	}
+	t.fullpath = fullpath
+
+	// adds metadata to enable users to filter logs by filename
+	t.tags = t.buildTailerTags()
+
+	log.Info("Opening ", t.fullpath)
+	f, err := openFile(t.fullpath)
+	if err != nil {
+		return err
+	}
+	filePos, _ := f.Seek(offset, whence)
+	f.Close()
+
+	t.readOffset = filePos
+	t.decodedOffset = filePos
+
+	return nil
+}
+
+func (t *Tailer) readAvailable() (err error) {
+	f, err := openFile(t.fullpath)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	st, err := f.Stat()
+	if err != nil {
+		log.Debugf("Error stat()ing file %v", err)
+		return err
+	}
+
+	sz := st.Size()
+	offset := t.GetReadOffset()
+	log.Debugf("Size is %d, offset is %d", sz, offset)
+	if sz == 0 {
+		log.Debug("File size now zero, resetting offset")
+		t.SetReadOffset(0)
+		t.SetDecodedOffset(0)
+	} else if sz < offset {
+		log.Debug("Offset off end of file, resetting")
+		t.SetReadOffset(0)
+		t.SetDecodedOffset(0)
+	}
+	f.Seek(t.GetReadOffset(), io.SeekStart)
+
+	inBuf := make([]byte, 4096)
+	for {
+		n, err := f.Read(inBuf)
```

Comment:
since we're using the same buffer (with the same underlying array) in each iteration of the loop, I think each iteration of the loop overwrites the buffer of the previous iteration. Downstream this will mutate the bytes that are being processed by the decoder, and lead to missed/duplicated/inconsistent logs.

Reply:
addressed in https://github.com/DataDog/datadog-agent/pull/5627